lib/selinux.c

```diff
@@ -98,6 +98,50 @@ int reset_selinux_file_context (void)
 	return 0;
 }
 
+/*
+ * check_selinux_permit - Checks whether SELinux grants the given operation
+ *
+ *     Returns 0 when permission is granted
+ *                    or something failed but running in permissive mode
+ */
+int check_selinux_permit(const char *perm_name)
+{
+	security_class_t passwd_class_id;
+	access_vector_t permission_id;
+
+	/* caller has checked for (is_selinux_enabled () > 0) already */
+
+	passwd_class_id = string_to_security_class ("passwd");
+	if (0 == passwd_class_id) {
+		fprintf (stderr,
+			    _("%s: can not find SELinux class 'passwd'\n"),
+			    Prog);
+		SYSLOG ((LOG_WARN, "can not find SELinux class 'passwd'"));
+		if (0 != security_getenforce ()) {
```

Comment:
Checking security_deny_unknown() would be more appropriate here, and is what libselinux selinux_check_access() does.

Reply:
switched to `selinux_check_access()`